botorch/acquisition/joint_entropy_search.py

```diff
@@ -62,42 +84,198 @@ def __init__(
+
+                # This equates to the JES version proposed by Hvarfner et. al.
+                if self.condition_noiseless:
+                    opt_noise = torch.full_like(
+                        self.optimal_outputs, MIN_INFERRED_NOISE_LEVEL)
+                    # conditional (batch) model of shape (num_models) x num_optima_per_model
+                    self.conditional_model = self.initial_model.condition_on_observations(
+                        X=self.initial_model.transform_inputs(self.optimal_inputs),
+                        Y=self.optimal_outputs,
+                        noise=opt_noise
+                    )
+                else:
+                    self.conditional_model = self.initial_model.condition_on_observations(
+                        X=self.initial_model.transform_inputs(self.optimal_inputs),
+                        Y=self.optimal_outputs,
+                    )
+                    
+        self.estimation_type = estimation_type
+        self.set_X_pending(X_pending)
 
     @concatenate_pending_points
     @t_batch_mode_transform()
     def forward(self, X: Tensor) -> Tensor:
         r"""Evaluates qLowerBoundJointEntropySearch at the design points `X`.
+        Args:
+            X: A `batch_shape x q x d`-dim Tensor of `batch_shape` t-batches with `q`
+            `d`-dim design points each.
+
+        Returns:
+            A `batch_shape`-dim Tensor of acquisition values at the given design
+            points `X`.
+        """
+        if self.estimation_type == "LB":
+            res = self._compute_lower_bound_information_gain(X)
+        elif self.estimation_type == "MC":
+            res = self._compute_monte_carlo_information_gain(X)
+        else:
+            raise ValueError(
+                f"Estimation type {self.estimation_type} is not valid."
+                f"Please specify any of {ESTIMATION_TYPES}"
+            )
+        return res
+
+    def _compute_lower_bound_information_gain(
+        self, X: Tensor, return_parts: bool = False
+    ) -> Tensor:
+        r"""Evaluates the lower bound information gain at the design points `X`.
+
+        Args:
+            X: A `batch_shape x q x d`-dim Tensor of `batch_shape` t-batches with `q`
+            `d`-dim design points each.
+
+        Returns:
+            A `batch_shape`-dim Tensor of acquisition values at the given design
+            points `X`.
+        """
+        tkwargs = {
+            "dtype": self.optimal_outputs.dtype,
+            "device": self.optimal_outputs.device,
+        }
+        initial_posterior = self.initial_model.posterior(X, observation_noise=True)
+        # need to check if there is a two-dimensional batch shape -
+        # the sampled optima appear in the dimension right after
+        batch_shape = X.shape[:-2]
+        sample_dim = len(batch_shape)
+        # We DISREGARD the additional constant term.
+        initial_entropy = 0.5 * torch.logdet(initial_posterior.mvn.covariance_matrix)
```

Comment:
The nice thing with this is that if the covariance matrix has special structure, thanks to linear_opeator's support for `__torch_function__` (see [here](https://github.com/cornellius-gp/linear_operator/blob/main/linear_operator/operators/_linear_operator.py#L1695-L1701) for the registration of the method)t his call will exploit that structure. 
```suggestion
        initial_entropy = 0.5 * torch.logdet(initial_posterior.mvn.lazy_covariance_matrix)
```

cc @gpleiss, @jacobrgardner